src/vm/tests/traits.rs

```diff
@@ -60,6 +62,63 @@ fn test_dynamic_dispatch_by_defining_trait(owned_env: &mut OwnedEnvironment) {
     }
 }
 
+fn test_dynamic_dispatch_pass_trait_nested_in_let(owned_env: &mut OwnedEnvironment) {
+    let dispatching_contract =
+        "(define-trait trait-1 (
+            (get-1 (uint) (response uint uint))))
+        (define-public (wrapped-get-1 (contract <trait-1>))
+            (let ((amount u0))
+              (internal-get-1 contract)))
+        (define-public (internal-get-1 (contract <trait-1>))
+            (ok u1))";
```

Comment:
I'm confused by this test -- does this test call `get-1` in `target_contract`?  Because it looks like `internal-get-1` just returns `(ok u1)` regardless of `contract`.

Reply:
we can make it so (this is already tested in other tests), the main point of this test is to ensure that the function `internal-get-1` can recursively retrieve the target_contract.
Before this patch, `test_dynamic_dispatch_pass_trait` is passing, but `test_dynamic_dispatch_pass_trait_nested_in_let` is not.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1644/commits/48f4bdf29356e109eea928027b6eb2b36e3884fc